src/middleware.py

```diff
@@ -1,8 +1,22 @@
 from fastapi import Request
+from src.models import SessionLocal, Department
+import contextvars
 
 
-async def print_hello(request: Request, call_next):
-    print("Hello")
+async def start_session(data: dict):
+    session_var = contextvars.ContextVar("session")
+    async with SessionLocal() as session:
+        with session.begin():
+            session_var.set(session)
+            obj = Department(**data)
+            session.add(obj)
+            session.flush()
+            session.refresh(obj)
+            session.expunge_all()
+            return obj
+
+
+async def reset_variable(request: Request, call_next, session_var):
     response = await call_next(request)
-    print("Goodbye")
+    session_var.reset()
     return response
```

Comment:
Look, after you created and set the session variable you should call request and then reset the var. In a single middleware. Imagine your `print hello` = create and begin session and set variable; your `print goodbye` = reset the var.